examples/escrow-cli/escrow_contract.rs

```diff
@@ -111,7 +118,9 @@ fn lookup_output<Record: TxORecord<EscrowDatum, ()>>(
         .iter()
         .find(|o| o.id() == id)
         .cloned()
-        .ok_or(format!("Couldn't find output with id: {}", id))
+        .ok_or(NauError::TxORecord(
```

Comment:
I think just adding a `SCLogic` error:

```
SCLogicError {
    Endpoint(Box<dyn Error>),
    Lookup(Box<dyn Error>),
}
```
prolly works for now. I mentioned using an associated type for `SCLogic` in #6, but this should be good enough.

I don't think this should be returning `NauError` though.

Reply:
It's just interesting since many of the errors this will be returning will be coming from the `txo_record` (TxORecordError).

Reply:
should `SCLogicError` be added in the `NauError` enum ? For the  `hit_endpoint` method in the `smart_contract` trait.

Reply:
Yeah. Exactly. What I'm envisioning is each trait has it's own error type:

`SmartContractTrait` -> `naumachia::error::Error` (this one is special because it's the top layer interface)
`TxORecord` -> `TxORecordError`
`SCLogic` -> `SCLogicError`
`ValidatorCode` -> `ValidatorCodeError`

`TxORecord`, `SCLogic`, and `ValidatorCode` are each traits inside our impl of `SmartContractTrait`, so they should each be variants of `SmartContract`'s Error type.

Errors should only be initialized within the impls for their traits. That helps us identify the source of the error; if we get a `TxORecordError`, we know it's in our impl for `TxORecord`, etc.

Reply:
Hope that helps you understand what I'm envisioning here.